### PR TITLE
Verify toml before write

### LINKED ISSUE #163: TOML-to-Header parser fails on edge-case
When calling:

```
./target/debug/imag-store --debug --rtp /tmp/ create -p /test entry -c content -h header=1
```

(from https://github.com/matthiasbeyer/imag/commit/57a12e1a0e90b6c10759d6ff93b37be35c41a7f1)

It generates the following file:

```
---header = "1"

[imag]
links = []
version = "0.1.0"
---
content 
```

Which, of course, fails to parse on `retrieve()`.

### PR DESCRIPTION
**Pull request by TheNeikos**
This closes #163

### CODE REVIEW

File: libimagstore/src/store.rs [hunk 1]
```diff
@@ -351,6 +357,16 @@ fn verify_header_consistency(t: Table) -> EntryResult<Table> {
     }
 }
 
+fn verify_header(t: &Table) -> Result<()> {
```
Comment (matthiasbeyer) [inline libimagstore/src/store.rs]: Pls move this to `EntryHeader::verify()`

**Comment by matthiasbeyer** [comment]
ACK.
